fix(event_handlers): handle lack of headers when using auto-compression feature

tbuatois opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->

**Issue number**: #1327 

## Summary

This PR add the ability use compress=True  in decorator without throwing an error for a request containing null headers

### Changes

* Add check for NoneType / null header in `get_header_value` function and return default if so
* Add corresponding unit test `test_compress_no_accept_encoding_null_headers`

### User experience

* Before
Error for a request with empty/null headers to a method with `compress=True`:
`'NoneType' object has no attribute 'items'", "errorType": "AttributeError"`  

That was due to an error from api_gateway.py at line 212:

```python
 if self.route.compress and "gzip" in (event.get_header_value("accept-encoding", "") or ""):
```

* After 

No more error, return default value as expected. 

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x ] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

tbuatois committed:
fix(event_handler): NoneType error for compress feature without headers

heitorlessa commented on aws_lambda_powertools/utilities/data_classes/common.py:
```diff
@@ -35,6 +35,10 @@ def get_header_value(
     if case_sensitive:
         return headers.get(name, default_value)
 
+    # If headers is NoneType, return default value
+    if not headers:
+        return default_value
+
```
```suggestion
    # If headers is NoneType, return default value
    if not headers:
        return default_value

    if case_sensitive:
        return headers.get(name, default_value)
```

Changing the order to prevent `case_sensitive=True` logic to kick in and fail on `AttributeError`, since `None` wouldn't have `.get`

heitorlessa reviewed (changes requested):
tiny change to address another potential error

tbuatois committed:
check order changes

Changing the order to prevent case_sensitive=True logic to kick in and fail on AttributeError, since None wouldn't have .get

Co-authored-by: Heitor Lessa <heitor.lessa@hotmail.com>
